### PR TITLE
Add tests for set_payment_method_title_for_email function

### PR DESCRIPTION
**Pull request by mdmoore**
#### Changes proposed in this Pull Request

This adds unit tests for the [`set_payment_method_title_for_email`](https://github.com/Automattic/woocommerce-payments/blob/ed2a18f74ab15ac1e0d7189f7904f8fc8c30d3ff/includes/payment-methods/class-upe-payment-gateway.php#L1178) function. A [recent issue](https://github.com/Automattic/woocommerce-payments/issues/7272) prompted some validation be added to the function and these tests cover those additions. These tests required mocking the `WC_Payments_Order_Service` class which was the only dependency of `UPE_Payment_Gateway` left to be mocked in the `UPE_Payment_Gateway_Test` tests.

#### Testing instructions

* Check that all tests are passing. Specifically, the `UPE_Payment_Gateway_Test` tests. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7331` or branch name `dev/7285-test-payment-method-title` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/7285-test-payment-method-title) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 3e5991609325a3f7adb53ca07410fe292189e889
- Build time: 2023-10-03 15:01:59 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: tests/unit/payment-methods/test-class-upe-payment-gateway.php [hunk 1]
```diff
@@ -1444,6 +1455,58 @@ public function test_correct_payment_method_title_for_order() {
...
+
+			$order = WC_Helper_Order::create_order();
+			$order->set_payment_method( 'woocommerce_payments' );
+			$order->set_payment_method_title( 'Popular Payment Methods' );
+
+			$this->mock_upe_gateway->set_payment_method_title_for_email( $order );
+			$this->assertEquals( 'WooPayments', $order->get_payment_method_title() );
+	}
+
+	public function test_set_payment_method_title_for_email_only_runs_for_legacy_upe() {
```
Comment (timur27) [inline tests/unit/payment-methods/test-class-upe-payment-gateway.php:1524]: I've left https://github.com/Automattic/woocommerce-payments/pull/7285#discussion_r1342647077 wondering if only legacy UPE should run this logic. Depending on the outcome, this test may or may not be removed.

**Review by timur27 — COMMENTED** [review]
Thanks Mike - I like how clean the tests are, without redundant constructions/logic. I've left one comment. 

One more thing is it looks like we currently have three use-cases tested: 
1. Payment method update skips for non-card gateways
2. Payment method update skips for non-legacy-UPE setting
3. Payment method is updated to the default title (`WooPayments`) if no payment method found

And we'd probably need to add yet another case: [actual title update](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/payment-methods/class-upe-payment-gateway.php#L1194-L1196) for card gateway to test e.g. `Popular payment methods` updates to `Visa debit card`.

**Comment by mdmoore** [comment]
> And we'd probably need to add yet another case: [actual title update](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/payment-methods/class-upe-payment-gateway.php#L1194-L1196) for card gateway to test e.g. `Popular payment methods` updates to `Visa debit card`.

Ah, yes... the core use case for the function 😃. Added in [27922c1](https://github.com/Automattic/woocommerce-payments/pull/7331/commits/27922c1eb94d960dccec243b68f993aa044e7b88).

File: tests/unit/payment-methods/test-class-upe-payment-gateway.php (hunk 1 above)
Comment (timur27) [reply tests/unit/payment-methods/test-class-upe-payment-gateway.php:1524]: Keeping the logic as is. See https://github.com/Automattic/woocommerce-payments/pull/7285#discussion_r1343711653 for more context.

**Review by timur27 — APPROVED** [review]
Looks great, Mike! Thank you! 🚢

**Pull request merged**